tests/eval_structure_generation/test_external_exclusions.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+import resources
+from resources import project_with_nested_external_dependencies
+
+from pytestarch.pytestarch import get_evaluable_architecture_for_module_objects
+
+
+def test_external_dependencies_present() -> None:
+    evaluable = get_evaluable_architecture_for_module_objects(
+        resources,
+        project_with_nested_external_dependencies,
+        exclude_external_libraries=False,
+    )
+    graph = evaluable._graph  # type: ignore
+
+    assert "logging" in graph
+    assert "logging.handlers" in graph
+
+
+def test_external_dependencies_removed_if_complete_match() -> None:
+    evaluable = get_evaluable_architecture_for_module_objects(
+        resources,
+        project_with_nested_external_dependencies,
+        exclude_external_libraries=False,
+        regex_external_exclusions=("logging",),
+    )
+    graph = evaluable._graph  # type: ignore
+
+    assert "logging" not in graph
+    assert "logging.handlers" not in graph
+
+
+def test_external_dependencies_removed_if_not_complete_match() -> None:
+    evaluable = get_evaluable_architecture_for_module_objects(
+        resources,
+        project_with_nested_external_dependencies,
+        exclude_external_libraries=False,
+        regex_external_exclusions=("logging$",),
+    )
+    graph = evaluable._graph  # type: ignore
```

Comment:
**question(non-blocking)**: why does `_graph` has to be private ? You need it in your tests. And in my experience, if we want to re-use the result of the analysis for something that your library is not designed for, we have to use it.

**suggestion(non-blocking)**: narrow your *ignore*. Here, you only want to `type: ignore[attr-defined]` because you bypass the rule to not access a protected member, but not all of them.

**thought**: I saw there is an issue about Ruff (#109), but this is related to type checking (typically MyPy). Maybe that's your IDE's plugins that notified you of it, but it may be interesting to type-check your codebase too. It is not a quick task usually, but the `pytestarch` code looks close to being fully type-checkable.